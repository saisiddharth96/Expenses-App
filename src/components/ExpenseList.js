import React from "react";
import { connect } from "react-redux";

import ExpenseListItem from "./ExpenseListItem.js";
import getVisibleExpenses from "../selectors/expenses.js";

export const ExpenseList = props => (
  <div>
    <h1>Expense List component</h1>
    {props.expenses.length === 0 ? (
      <p>No expenses yet! Please add some</p>
    ) : (
      props.expenses.map(expense => (
        <ExpenseListItem
          key={expense.id}
          description={expense.description}
          amount={expense.amount}
          createdAt={expense.createdAt}
          id={expense.id}
          dispatch
        />
      ))
    )}
  </div>
);

const mapStateToProps = state => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
