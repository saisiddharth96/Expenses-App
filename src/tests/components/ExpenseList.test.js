import React from "react";
import { shallow } from "enzyme";
import { ExpenseList } from "../../components/ExpenseList.js";
import expenses from "../fixtures/expenses.js";

test("should render ExpenseList with expenses", () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpenseList without expenses", () => {
  const wrapper = shallow(<ExpenseList expenses={[]}/>);
  expect(wrapper).toMatchSnapshot();
});
