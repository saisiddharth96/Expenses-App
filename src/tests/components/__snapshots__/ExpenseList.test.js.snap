// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <h1>
    Expense List component
  </h1>
  <ExpenseListItem
    amount={12}
    createdAt={0}
    description="Gum"
    dispatch={true}
    id="1"
    key="1"
  />
  <ExpenseListItem
    amount={500}
    createdAt={-345600000}
    description="Rent"
    dispatch={true}
    id="2"
    key="2"
  />
  <ExpenseListItem
    amount={300}
    createdAt={345600000}
    description="Water bill"
    dispatch={true}
    id="3"
    key="3"
  />
</div>
`;

exports[`should render ExpenseList without expenses 1`] = `
<div>
  <h1>
    Expense List component
  </h1>
  <p>
    No expenses yet! Please add some
  </p>
</div>
`;
